Use expireAt as index (#11)
http-rs/async-mongodb-session: README.md

$ cargo add async-mongodb-session
```

## Overview
This library allow you to utilises the document expiration feature based on a [specified number of seconds](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-after-a-specified-number-of-seconds) or in a [specific clock time](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time) supported by mongodb to expire the session.

> 

Reviewer: I was thinking something along the lines of
"The **specified number of seconds** approach is designed to enable the session time out to be managed at the mongodb layer. This approach provides a globally consistent session timeout across multiple processes but has the downside that all services using the same session collection **must** use the same timeout value.

The **specific clock time** clock time approach is where you require more flexibility on your session timeouts such as a different session timeout per running service or you would prefer to manage the session time out at the process level. This is more flexible but might lead to some perceived inconsistency in session timeout depending on your upgrade/rollout strategy."

Author: Hi @No9, Neat!! If you want I can add those lines. 👍

Reviewer: @pepoviola If you're OK with it then lets add it inbetween the content that already there

Author: Hi @No9, added 👍 . Thanks for the help!

$ cargo add async-mongodb-session
```

> ## Overview
> This library allow you to utilises the document expiration feature based on a [specified number of seconds](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-after-a-specified-number-of-seconds) or in a [specific clock time](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time) supported by mongodb to expire the session.

Reviewer 2: ```suggestion
## Overview
This library uses the document expiration feature based on a [specified number of seconds](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-after-a-specified-number-of-seconds) or a [specific clock time](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time) supported by mongodb to expire the session.
```

The specified number of seconds approach is designed to enable the session time out to be managed at the mongodb layer. This approach provides a globally consistent session timeout across multiple processes but has the downside that all services using the same session collection must use the same timeout value.

> The specific clock time clock time approach is where you require more flexibility on your session timeouts such as a different session timeout per running service or you would prefer to manage the session time out at the process level. This is more flexible but might lead to some perceived inconsistency in session timeout depending on your upgrade/rollout strategy.

Reviewer 2: i'm curious about the word "perceived" — is it actually consistent?

> The specific clock time clock time approach is where you require more flexibility on your session timeouts such as a different session timeout per running service or you would prefer to manage the session time out at the process level. This is more flexible but might lead to some perceived inconsistency in session timeout depending on your upgrade/rollout strategy.

Reviewer 2: ```suggestion
The specific clock time approach is for use cases when you require more flexibility on your session timeouts such as a different session timeout per running service or you would prefer to manage the session time out at the process level. This is more flexible but might lead to some perceived inconsistency in session timeout depending on your upgrade/rollout strategy.
```

## Configuration

This library utilises the document [expiry feature](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time) in mongodb to expire the session at the expiry.

The management of the expiry feature fits into the 12 factor [admin process definintion](https://12factor.net/admin-processes) so it's recommended to use an process outside of your web application to manage the expiry parameters.

A `expireAt` property is available on the root of the session document IFF the session expire is set. Note that  [async-session doesn't set by default](https://github.com/http-rs/async-session/blob/main/src/session.rs#L98).

To enable this [expiry feature](https://docs.mongodb.com/manual/tutorial/expire-data/#expire-documents-at-a-specific-clock-time) at `index` for `expireAt` is created when the client `connet`. Allowing then to use this property from the session to set the expiration of the document.

> 

Reviewer: I think we should keep both approaches in the README as well as state when you would use one over the other.

Then the script to create the expiry would be:
```
use db_name;
db.coll_session.createIndex( { "created": 1 } , { expireAfterSeconds: 300 } );
```

If you wish to redefine the session duration then the index must be dropped first using:
```
use db_name;
db.coll_session.dropIndex( { "created": 1 })
db.coll_session.createIndex( { "created": 1 } , { expireAfterSeconds: 300 } );
```

> Other way to set create the index is using  `index_on_created` passing the amount of seconds to expiry after the session.

Reviewer 2: ```suggestion
Another way to set create the index is to use  `index_on_created`, passing the amount of seconds to expiry after the session.
```
